File: 백준/Bronze/2798. 블랙잭/블랙잭.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args){

        Scanner scanner = new Scanner(System.in);
        int n,m;
        int sum=0;
        n=scanner.nextInt(); // n장
        m= scanner.nextInt(); //m 넘지 않게
        int card[]= new int[n];

        for(int i=0; i<n; i++){
            card[i]= scanner.nextInt();
        }
        
        int result=0;
        Label:
        for(int i=0; i<n-2; i++){
            for(int j=i+1; j<n-1; j++){
                for(int h=j+1; h<n; h++){
                    sum=card[i]+card[j]+card[h];
                    if(m == sum){
                        result=m;
                        break Label;
                    }
                    if(sum<m && sum>result){
                        result=sum;
                    }
                }
            }
        }
        System.out.println(result);
    }
}

File: 백준/Bronze/2798. 블랙잭/블랙잭.py
n, m = map(int, input().split())

result = 0
card = list(map(int, input().split()))

for i in range(len(card)):
    for j in range(i+1, len(card)):
        for z in range(j+1, len(card)):
            if(card[i]+card[j]+card[z] == m):
                result = m
                break
            elif(card[i]+card[j]+card[z] < m):
                result = max(result, card[i]+card[j]+card[z])
        else:
            continue
        break
    else:
        continue
    break
print(result)